crates/router/src/connector/aci/transformers.rs

```diff
@@ -59,6 +71,162 @@ pub enum PaymentDetails {
+    type Error = Error;
+    fn try_from(wallet_data: &api_models::payments::WalletData) -> Result<Self, Self::Error> {
+        let payment_data = match wallet_data {
+            api_models::payments::WalletData::MbWay(data) => Self::Wallet(Box::new(WalletPMData {
+                payment_brand: PaymentBrand::Mbway,
+                account_id: Some(data.telephone_number.clone()),
+            })),
+            api_models::payments::WalletData::AliPay { .. } => {
+                Self::Wallet(Box::new(WalletPMData {
+                    payment_brand: PaymentBrand::AliPay,
+                    account_id: None,
+                }))
+            }
+            _ => Err(errors::ConnectorError::NotImplemented(
+                "Payment method".to_string(),
+            ))?,
+        };
+        Ok(payment_data)
+    }
+}
+
+impl
+    TryFrom<(
+        &types::PaymentsAuthorizeRouterData,
+        &api_models::payments::BankRedirectData,
+    )> for PaymentDetails
+{
+    type Error = Error;
+    fn try_from(
+        value: (
+            &types::PaymentsAuthorizeRouterData,
+            &api_models::payments::BankRedirectData,
+        ),
+    ) -> Result<Self, Self::Error> {
+        let (item, bank_redirect_data) = value;
+        let payment_data = match bank_redirect_data {
+            api_models::payments::BankRedirectData::Eps { .. } => {
+                Self::BankRedirect(Box::new(BankRedirectionPMData {
+                    payment_brand: PaymentBrand::Eps,
+                    bank_account_country: Some(api_models::enums::CountryAlpha2::AT),
+                    bank_account_bank_name: None,
+                    bank_account_bic: None,
+                    bank_account_iban: None,
+                    billing_country: None,
+                    merchant_customer_id: None,
+                    merchant_transaction_id: None,
+                    customer_email: None,
+                }))
+            }
+            api_models::payments::BankRedirectData::Giropay {
+                bank_account_bic,
+                bank_account_iban,
+                ..
+            } => Self::BankRedirect(Box::new(BankRedirectionPMData {
+                payment_brand: PaymentBrand::Giropay,
+                bank_account_country: Some(api_models::enums::CountryAlpha2::DE),
+                bank_account_bank_name: None,
+                bank_account_bic: bank_account_bic.clone(),
+                bank_account_iban: bank_account_iban.clone(),
+                billing_country: None,
+                merchant_customer_id: None,
+                merchant_transaction_id: None,
+                customer_email: None,
+            })),
+            api_models::payments::BankRedirectData::Ideal { bank_name, .. } => {
+                Self::BankRedirect(Box::new(BankRedirectionPMData {
+                    payment_brand: PaymentBrand::Ideal,
+                    bank_account_country: Some(api_models::enums::CountryAlpha2::NL),
```

Comment:
do not hardcode the country get it from input, do api_contract_change if necessary